waiter/src/waiter/util/http_utils.clj

```diff
@@ -209,3 +209,13 @@
   (or (= http-503-service-unavailable (:status response))
       (and (grpc? (:headers request) (:client-protocol request))
            (= "14" (get-in response [:headers "grpc-status"])))))
+
+(defn browser-request?
+  "Looks at the user-agent header to determine if the request came from a browser.
+   This is useful, e.g., determining whether the client should be asked to handle redirects.
+   Current implementation checks for the presence of chrome or mozilla based on common user-agents:
+   https://www.networkinghowtos.com/howto/common-user-agent-list/"
+  [request]
+  (when-let [user-agent (some-> request (get-in [:headers "user-agent"]) str/lower-case)]
+    (or (str/includes? user-agent "mozilla")
+        (str/includes? user-agent "chrome"))))
```

Comment:
Only FF and Chrome?  Do we have to support other browsers for mobile devices?

Reply:
"Chome" is actually redundant here - every modern browser includes the Mozilla string at the start of it's user-agent

Reply:
Discussed this offline with @blevz and he feels we should keep the `chrome` string in case Chrome changes their user-agent in the future.

Reply:
I was thinking something along the lines of:

Checking for "mozilla" and "chrome" our best heuristic for determining whether something is a browser.